logstash_async/formatter.py

```diff
@@ -133,7 +133,9 @@ def _format_timestamp(self, time_):
 
     # ----------------------------------------------------------------------
     def _get_record_fields(self, record):
-        return {k: self._value_repr(v) for k, v in record.__dict__.items()}
+        return {k: self._value_repr(v)
```

Comment:
Nice!
Since we removed now all unwanted fields from the record, we could skip trying to remove them again in `_remove_excluded_fields` by only passing `extra_fields` to the method.
Or maybe we incoporate the field removing code into `_get_extra_fields` as it remains the only place where the filtering is necessary yet.

Reply:
Yes, I thought of that. However, subclasses may depend on `_remove_excluded_fields` to be called. 

Another optimization would be to fold the `_get_record_fields`, `_get_extra_fields` and `_remove_excluded_fields` into one method: It iterates over `record.__dict__` and builds a `dict` that can be `message.update`d in the `format` method. This would eliminate some iterations, checks and eventually `del` calls as well.
However, this would be a larger break in API and behaviour.

Reply:
Good point about API breakage.

Maybe it's worth for these changes to bump to 3.0 and announce the API changes, so people will notice.